docs/adr/0001-update-logins-api.md

```diff
@@ -0,0 +1,77 @@
+    - [get_by_base_domain](https://github.com/mozilla/application-services/blob/1248a352cb2701b92395f2783bee8a88d18de455/components/logins/src/db.rs#L279)
+    - [get_by_id](https://github.com/mozilla/application-services/blob/1248a352cb2701b92395f2783bee8a88d18de455/components/logins/src/db.rs#L330)
+    - [check_valid_with_no_dupes](https://github.com/mozilla/application-services/blob/1248a352cb2701b92395f2783bee8a88d18de455/components/logins/src/db.rs#L663)
+    - [potential_dupes_ignoring_username](https://github.com/mozilla/application-services/blob/1248a352cb2701b92395f2783bee8a88d18de455/components/logins/src/db.rs#L721)
+    - [import_multiple](https://github.com/mozilla/application-services/blob/1248a352cb2701b92395f2783bee8a88d18de455/components/logins/src/db.rs#L455)
+
+      **Note:** Functions related to sync have been omitted as it’s assumed they’ll have access to decrypted data.
+
+  The proposed changes will reduce the above list of functions to the following:
+    - `save_or_update`
+      - This function will replace `add` and `update`
+    - `import_multiple`
+      - This function will be removed once the Fennec to Fenix migration period ends
+
+  Both the `potential_dupes_ignoring_username` and `check_valid_with_no_dupes` functions will removed from the API. Neither function is called in Firefox iOS. Android Components uses both to provide validation and de-duping before logins are added or updated. We can eliminate the need to externalize these functions by replicating this logic in the new `save_or_update` function.
+
+  If we exclude the encrypted `username` and `password` fields from [the Nul check](https://github.com/mozilla/application-services/blob/1248a352cb2701b92395f2783bee8a88d18de455/components/logins/src/login.rs#L446) in the validate and fixup logic, we can eliminate the need for the `get_all`, `get_by_base_domain`, and `get_by_id` functions to have the encryption key.
+
+* Pros
+  * Improves the logins API for consumers by combining add/update functionality (see [#3899](https://github.com/mozilla/application-services/issues/3899) for details)
+  * Removes redundant validation and de-duping logic in consumer code
+  * Uses the same encryption model as autofill so there is consistency in our approaches
+* Cons
+  * Potentially weakens the fixup and validation logic to either cover fewer logins fields or fewer functions
+  * Breaks the logins API and requires consumer code changes
+  * Requires consumer code to both encrypt logins fields _and_ pass the encryption key when calling either `save_or_update` and `import_multiple`
+
+
+### Option 2 - Implement a different encryption approach
+
+(Judging from the comments in our tickets I don't think we've identified a viable alternative but I'm adding this section here in case I'm mistaken and there are other options to document.)
```

Comment:
If I'm wrong here, please let me know and/or feel free to add other options.

Reply:
I think this would be something like "let the component hold on to the key and therefore make the api unavailable if the key isn't held" - which is roughly what we have now. There's no reason we couldn't stick with the same fundamental model we have now even with per-field encryption and sqlite.

(I'm inclined to think this model defeats alot of the purpose in an encrypted store, but doesn't defeat all of it, so I guess we should keep it on the table)

Reply:
I'm assuming this would mean keeping the encryption key in the db "open" functions , require a code review, and still allow the same consolidation of API functions.